components/kyma-environment-broker/internal/avs/delegator.go

```diff
@@ -88,17 +93,54 @@ func (del *Delegator) AddTags(logger logrus.FieldLogger, operation internal.Prov
 			errMsg := "cannot add tags to AVS evaluation (temporary)"
 			logger.Errorf("%s: %s", errMsg, err)
 			retryConfig := evalAssistant.provideRetryConfig()
-			op, duration, err := del.operationManager.RetryOperation(operation, errMsg, retryConfig.retryInterval, retryConfig.maxTime, logger)
+			op, duration, err := del.provisionManager.RetryOperation(operation, errMsg, retryConfig.retryInterval, retryConfig.maxTime, logger)
 			return op, duration, err
 		default:
 			errMsg := "cannot add tags to AVS evaluation"
 			logger.Errorf("%s: %s", errMsg, err)
-			op, duration, err := del.operationManager.OperationFailed(operation, errMsg)
+			op, duration, err := del.provisionManager.OperationFailed(operation, errMsg)
 			return op, duration, err
 		}
 	}
 
-	updatedOperation, d = del.operationManager.UpdateOperation(operation)
+	updatedOperation, d = del.provisionManager.UpdateOperation(operation)
+
+	return updatedOperation, d, nil
+}
+
+func (del *Delegator) SetStatus(logger logrus.FieldLogger, operation internal.UpgradeKymaOperation, evalAssistant EvalAssistant, status string) (internal.UpgradeKymaOperation, time.Duration, error) {
+	logger.Infof("starting the SetStatus to avs internal id [%d]", operation.Avs.AvsEvaluationInternalId)
+	var updatedOperation internal.UpgradeKymaOperation
+	d := 0 * time.Second
+
+	evalId := evalAssistant.GetEvaluationId(operation.Avs)
+	currentStatus := evalAssistant.GetEvalStatus(operation.Avs)
```

Comment:
NOTE: Avs lifecycle data in `UpgradeKymaOperation` will be present after https://github.com/kyma-project/control-plane/pull/437

Reply:
The current logic relies on migrations, and this shouldn't be the case. To support the scenario when the migrations are not working, it's possible to have default states (`Maintenance` on upgrade init and `Active` on finish). The worst-case scenario is that we'll set `Active` on upgrade finish for a monitor that had `originalState != Active` before the upgrade, due to inability to preserve states. However, this edge-case will be resolved as soon as #437 is merged.